libcxx/include/__type_traits/aligned_storage.h

```diff
@@ -88,9 +71,7 @@ struct __find_max_align<__type_list<_Hp, _Tp>, _Len>
 
 template <size_t _Len, size_t _Align = __find_max_align<__all_types, _Len>::value>
 struct _LIBCPP_DEPRECATED_IN_CXX23 _LIBCPP_TEMPLATE_VIS aligned_storage {
-  typedef typename __find_pod<__all_types, _Align>::type _Aligner;
-  union type {
-    _Aligner __align;
+  union _ALIGNAS(_Align) type {
```

Comment:
The patch description says:

> The main template of aligned_storage is only ever used when we have extremely overaligned types (> 16384), so we effectively only ever use the specializations currently. This means that we only instantiate the main template for overaligned types. Instead of doing this dance, we can just define the main template to use _ALIGNAS, just like the specializations. This makes the implementation of aligned_storage significantly less confusing.

I got a bit confused by the first sentence, which seems to suggest that this patch is a functionality change. However, if I understand correctly, the bottom line is that this patch just ditches the specializations and always uses `_ALIGNAS`, without changing any functionality. I assume the reason for having specializations in the first place must have been for compile-time performance?

Reply:
The farthest I can go also dates back to the initial SVN import, unfortunately.